core/src/main/java/org/mule/runtime/core/internal/util/CompositeClassLoader.java

```diff
@@ -35,15 +41,57 @@
 
   private List<ClassLoader> delegates;
 
-  public CompositeClassLoader(ClassLoader first, ClassLoader... others) {
-    delegates = new ArrayList<>();
+  private static final Cache<Pair<ClassLoader, ClassLoader>, CompositeClassLoader> cache = Caffeine.newBuilder()
```

Comment:
Can we verify how this works on multiple and stressing undeployments? Maybe setting weakreferences?